### PR TITLE
Fixes for PBS and more

### PR DESCRIPTION
**Pull request by askask**

### CODE REVIEW

File: src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/ClusterJobManager.groovy [hunk 1]
```diff
@@ -32,8 +33,15 @@ abstract class ClusterJobManager<C extends Command> extends BatchEuphoriaJobMana
...
-            logger.warning(e.stackTrace.join("\n"))
+            log.warn(e.message)
+            List<StackTraceElement> stel = []
+            for (StackTraceElement element : e.stackTrace) {
+                stel.add(element)
+                if (element.toString().contains("JobManager")) {
+                    break
+                }
+            }
+            log.warn(stel.join("\n"))
```
[the remark this answers is not in the document]
Comment (vinjana) [reply src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/ClusterJobManager.groovy]: O.k. Unfortunately, I do not know a better solution. You to achieve the same you could use the stream API for collections, but I doubt that's less complex.

File: src/main/groovy/de/dkfz/roddy/execution/jobs/GenericJobInfo.groovy [hunk 2]
```diff
@@ -26,22 +26,22 @@ class GenericJobInfo {
     File tool
     BEJobID jobID
 
-    LocalDateTime submitTime;
-    LocalDateTime startTime;
-    LocalDateTime endTime;
-    LocalDateTime eligibleTime; // when all conditions like job dependencies full filled, it is qu
+    LocalDateTime submitTime
+    LocalDateTime eligibleTime // when all conditions like job dependencies full filled, it is qu
```
[the remark this answers is not in the document]
Comment (askask) [reply src/main/groovy/de/dkfz/roddy/execution/jobs/GenericJobInfo.groovy]: Fixed

File: src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/GridEngineBasedJobManager.groovy [hunk 3]
```diff
@@ -171,39 +157,34 @@ abstract class GridEngineBasedJobManager<C extends Command> extends ClusterJobMa
...
             BEJobID jobID
             try {
                 jobID = new BEJobID(jobIdRaw)
             } catch (Exception exp) {
                 throw new BEException("Job ID '${jobIdRaw}' could not be transformed to BEJobID ")
             }
 
-            List<String> jobDependencies = it["depend"] ? (it["depend"] as  String).find("afterok.*")?.findAll(/(\d+).(\w+)/) { fullMatch, beforeDot, afterDot -> return beforeDot } : null
-            GenericJobInfo gj = new GenericJobInfo(it["Job_Name"] as String ?: null, null, jobID, null, jobDependencies)
+            List<String> jobDependencies = catchAndLogExceptions { getJobDependencies(job["depend"] as String) }
```
[the remark this answers is not in the document]
Comment (askask) [reply src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/GridEngineBasedJobManager.groovy]: Good idea, fixed

File: src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/GridEngineBasedJobManager.groovy [hunk 4]
```diff
@@ -221,58 +202,80 @@ abstract class GridEngineBasedJobManager<C extends Command> extends ClusterJobMa
...
+        String fileName
+        if (s.startsWith("/")) {
+            fileName = s
         } else if (s =~ /^[\w-]+:\//) {
-            return new File(s.replaceAll(/^[\w-]+:/, ""))
+            fileName = s.replaceAll(/^[\w-]+:/, "")
         } else {
             return null
         }
+        new File(fileName.replace("\$PBS_JOBID", jobId))
```
[the remark this answers is not in the document]
Comment (askask) [reply src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/GridEngineBasedJobManager.groovy]: Fixed

File: src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/GridEngineBasedJobManager.groovy [hunk 5]
```diff
@@ -221,58 +202,80 @@ abstract class GridEngineBasedJobManager<C extends Command> extends ClusterJobMa
...
+            if (job["etime"])  // The time that the job became eligible to run, i.e. in a queued state while residing in an execution queue.
+                gj.setEligibleTime(parseTime(job["etime"] as String))
 
             queriedExtendedStates.put(jobID, gj)
         }
         return queriedExtendedStates
     }
 
-    private static File getQstatFile(String s) {
+    private static List<String> getExecutionHosts(String s) {
```
[the remark this answers is not in the document]
Comment (askask) [reply src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/GridEngineBasedJobManager.groovy]: Fixed

File: src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/GridEngineBasedJobManager.groovy [hunk 6]
```diff
@@ -221,58 +202,80 @@ abstract class GridEngineBasedJobManager<C extends Command> extends ClusterJobMa
...
+                .unique()
+    }
+
+    private static List<String> getJobDependencies(String s) {
         if (!s) {
             return null
-        } else if (s.startsWith("/")) {
-            return new File(s)
+        }
+        s.split(",")
```
[the remark this answers is not in the document]
Comment (askask) [reply src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/GridEngineBasedJobManager.groovy]: Fixed

**Pull request merged**